dataprofiler/profilers/profile_builder.py

```diff
@@ -1223,17 +1231,85 @@ def _min_col_samples_used(self):
         i.e. every column in the Profile was read up to this row (possibly
         further in some cols)
         """
-        return min([self._profile[col].sample_size
-                    for col in self._profile], default=0)
+        return min([col.sample_size for col in self._profile], default=0)
 
     @property
     def _min_sampled_from_batch(self):
         """
         Calculates and returns the number of rows that were completely sampled
         in the most previous batch
         """
-        return min([self._profile[col]._last_batch_size
-                    for col in self._profile], default=0)
+        return min([col._last_batch_size for col in self._profile], default=0)
+
+    @property
+    def is_initialized(self):
+        """
+        Determines if profiler has been initialized with data
+        """
+        return self.total_samples > 0
+
+    def _get_col_name_from_idx(self, idx):
+        """
+        Determine which column name corresponds to a _profile index
+        """
+        for col_name in self._col_name_to_idx:
+            if idx in self._col_name_to_idx[col_name]:
+                return col_name
+
+    @staticmethod
+    def _get_and_validate_schema(schema1, schema2, strict=False):
+        """
+        Validate compatibility between schema1 and schema2 and return a dict
+        mapping indices in schema1 to their corresponding indices in schema2.
+        In __add__: want to map self _profile idx -> other _profile idx
+        In _update_profile_from_chunk: want to map data idx -> _profile idx
+
+        :param schema1: a column name to index mapping
+        :type schema1: Dict[str, list[int]]
+        :param schema2: a column name to index mapping
+        :type schema2: Dict[str, list[int]]
+        :param strict: whether or not to strictly match (__add__ case)
+        :type strict: bool
+        :return: a mapping of indices in schema1 to indices in schema2
+        :rtype: Dict[int, int]
+        """
+
+        one_schema_empty = (len(schema1) == 0 and len(schema2) > 0) \
+                           or (len(schema1) > 0 and len(schema2) == 0)
+        # In the case of __add__ with one of the schemas not initialized
+        if strict and one_schema_empty:
+            raise ValueError("Cannot merge profiles due to schema mismatch")
```

Comment:
What is the case for both being empty?

Reply:
There was a case where merging an initialized schema with unitinitialized should have raised an error, but didn't due to the logic for the uninitialized profiler in update_from_chunk case. So this ensures that in the merge case, this is no longer a blind spot. And the case for both being empty is 2 uninitialized profiles being merged, which I am unsure on what the correct behavior is, but if it is incorrect then I can make this 1 or statement, I just didn't know if both being empty was acceptable or not for merge.

Reply:
This issue will be saved for a later refactor. We will want to allow merging empty profiles in the future.